plugins/woocommerce/tests/e2e-pw/tests/merchant/product-edit.spec.js

```diff
@@ -1,64 +1,187 @@
+			await page.getByLabel( 'Product name' ).fill( newProduct.name );
+		} );
+
+		await test.step( 'edit the product description', async () => {
+			await page.locator( '#content-html' ).click(); // text mode to work around iframe
+			await page
+				.locator( '.wp-editor-area' )
+				.first()
+				.fill( newProduct.description );
+		} );
+
+		await test.step( 'edit the product price', async () => {
+			await page
+				.getByLabel( 'Regular price ($)' )
+				.fill( newProduct.regularPrice );
+			await page
+				.getByLabel( 'Sale price ($)' )
+				.fill( newProduct.salePrice );
+		} );
+
+		await test.step( 'publish the updated product', async () => {
+			await page.locator( '#publish' ).click();
 		} );
-		await api.delete( `products/${ productId }`, {
-			force: true,
+
+		await test.step( 'verify the changes', async () => {
+			await expect( page.getByLabel( 'Product name' ) ).toHaveValue(
+				newProduct.name
+			);
+			await expect(
+				page.locator( '.wp-editor-area' ).first()
+			).toContainText( newProduct.description );
+			await expect( page.getByLabel( 'Regular price ($)' ) ).toHaveValue(
+				newProduct.regularPrice
+			);
+			await expect( page.getByLabel( 'Sale price ($)' ) ).toHaveValue(
+				newProduct.salePrice
+			);
 		} );
 	} );
 
-	test( 'can edit a product and save the changes', async ( { page } ) => {
-		await page.goto( `wp-admin/post.php?post=${ productId }&action=edit` );
+	test( 'can bulk edit products', async ( { page, products } ) => {
+		await page.goto( `wp-admin/edit.php?post_type=product` );
 
-		// make some edits
-		await page.locator( '#title' ).fill( 'Awesome product' );
-		await page.locator( '#content-html' ).click(); // text mode to work around iframe
-		await page
-			.locator( '.wp-editor-area >> nth=0' )
-			.fill( 'This product is pretty awesome' );
-		await page.locator( '#_regular_price' ).fill( '100.05' );
+		const regularPriceIncrease = 10;
+		const salePriceDecrease = 10;
+		const stockQtyIncrease = 10;
 
-		// publish the edits
-		await page.locator( '#publish' ).click();
+		await test.step( 'select and bulk edit the products', async () => {
+			for ( const product of products ) {
+				await page.getByLabel( `Select ${ product.name }` ).click();
+			}
 
-		// verify the changes saved
-		await expect( page.locator( '#title' ) ).toHaveValue(
-			'Awesome product'
-		);
-		await expect(
-			page.locator( '.wp-editor-area >> nth=0' )
-		).toContainText( 'This product is pretty awesome' );
-		await expect( page.locator( '#_regular_price' ) ).toHaveValue(
-			'100.05'
-		);
+			await page
+				.locator( '#bulk-action-selector-top' )
+				.selectOption( 'Edit' );
+			await page.locator( '#doaction' ).click();
+
+			await expect(
+				await page.locator( '#bulk-titles-list li' ).count()
+			).toEqual( products.length );
+		} );
+
+		await test.step( 'update the regular price', async () => {
+			await page
+				.locator( 'select[name="change_regular_price"]' )
```

Comment:
Can we use a `page.getByRole()` for some of these as well?

Reply:
Actually no, `getByRole` was not an option in this case.